src/ContactModels/include/BipedalLocomotionControllers/ContactModels/ContinuousContactModel.h

```diff
@@ -0,0 +1,141 @@
+    iDynTree::Transform m_nullForceTransform; /**< Homogeneous transformation corresponding to a
+                                                 null contact wrench */
+    iDynTree::Twist m_twist; /**< Twist expressed in mixed representation of the frame placed in the
+                                center of the contact surface */
+
+    double m_springCoeff; /**< Spring coefficient associated to the model */
+    double m_damperCoeff; /**< Damper coefficient associated to the model */
+
+    double m_length; /**< Length of the rectangular contact surface */
+    double m_width; /**< Width of the rectangular contact surface */
+
+    /**
+     * Evaluate the contact wrench given a specific contact model
+     */
+    void computeContactWrench() final;
+
+    /**
+     * Evaluate the Autonomous System Dynamics of the derivative of a specific contact model
+     */
+    void computeAutonomousDynamics() final;
+
+    /**
+     * Evaluate the Control Matrix of the derivative of a specific contact model
+     */
+    void computeControlMatrix() final;
+
+    /**
+     * Set the parameters cannot change
+     * @param state std::unordered_map containing the mutable parameters of the system. To compute the contact
+     * wrench between two system (i.e. a link of a robot and the environment) the following data has
+     * to be available. If not an std::runtime_error is thrown.
+     *   - \a length (double): length in meters associated to the model
+     *   - \a width (double): width in meters associated to the model
+     */
+    void setImmutableParameters(const std::unordered_map<std::string, std::any>& parameters) final;
+
+public:
+
+    /**
+     * Constructor. It instantiate the value of the mutable parameters and the immutable parameters.
+     * @param immutableParameters std::unordered_map containing the immutable parameters of the
+     * system. The list of the required parameters could be found in @ref
+     * BipedalLocomotionController::ContactModels::setImmutableparameters
+     * @param mutableParameters std::unordered_map containing the mutable parameters of the
+     * system. The list of the required parameters could be found in @ref
+     * BipedalLocomotionController::ContactModels::setMutableparameters
+     */
+    ContinuousContactModel(const std::unordered_map<std::string, std::any>& immutableParameters,
+                           const std::unordered_map<std::string, std::any>& mutableParameters);
+
+    /**
+     * Constructor. It instantiate the value of the mutable parameters and the immutable parameters.
+     * @param parameters std::unordered_map containing the immutable and mutable parameters of the
+     * system. The list of the required parameters could be found in @ref
+     * BipedalLocomotionController::ContactModels::setMutableparameters
+     */
+    ContinuousContactModel(const std::unordered_map<std::string, std::any>& parameters);
```

Comment:
It sounds good. In this case, we should rethink the implementation of the class.
For instance the constructor of the `ContinuousContactModel` should become
```cpp
ContinuousContactModel::ContinuousContactModel()
     : m_twist(iDynTree::Twist::Zero())
     , m_frameTransform(iDynTree::Transform::Identity())
     , m_nullForceTransform(iDynTree::Transform::Identity())
{
     // reset matrix and vectors
     m_autonomousDynamics.zero();
     m_controlMatrix.zero();
}
```

an initialize method should be implemented 
```cpp
template <class Derived>
bool ContinuousContactModel::initialize(std::waeak_ptr<IParameterHanlder<derived> hanlder)
{
     // get the parameters
     return true;
}
```

At this point, I would avoid using the `std::unordered_map<std::string, std::any>` for passing not only the parameters but also the state.  In this case the implementation of the other function will be required. i.e. 

```cpp
void ContinuousContactModel::setNullForceFrameTransform(const iDynTree::Transform &transform)
{
    m_nullForceTransform = transform;
}
```

In this way, we also go to the direction suggested by @traversaro  in https://github.com/dic-iit/bipedal-locomotion-controllers/pull/7#issuecomment-573560373

Reply:
Honestly, I don't see anything wrong in using the map. I would say it would be better to avoid having templates around. It could unnecessarily complexify the interface.

To this end, you could use an alias so that you are free to change the type of handler without problems in the future. Or a new derived class specific to the ``COntinuosContactModel``, something like ``ContinuosContactModelParameters`` which inherit from ``IParametersHandler``.

Reply:
> Honestly, I don't see anything wrong in using the map. I would say it would be better to avoid having templates around. It could unnecessarily complexify the interface.

I agree.

Reply:
I know that having a template function may complexify the interface however is the only way to use the `ContinuousContactModel` with a different implementation of the handler. Till now we have only one implementation of the handler but hopefully, in the future, a `toml` implementation will be added #26 
I  really would like to avoid using CRPT in `IParameterHandler` but I don't know to update the parameter library.

Reply:
I was thinking that at some point you would use the `IParameterHandler` to convert to `std::unordered_map<std::string, std::any>`, but perhaps it is indeed to complex.

Reply:
I think that this comment has been addressed thanks to:
- https://github.com/dic-iit/bipedal-locomotion-controllers/commit/01d406c794e872a75eb967e3fd7f5236761cf955
- https://github.com/dic-iit/bipedal-locomotion-controllers/pull/7/commits/7a64ab0605ad7d7383b02f33dcd2ce0f66c2f30c